packages/eslint-plugin/tests/rules/strict-boolean-expressions.test.ts

```diff
@@ -965,6 +984,56 @@ if (y) {
       ],
     }),
 
+    // nullable enum in boolean context
+    {
+      options: [{ allowNullableEnum: false }],
+      code: `
+        enum ExampleEnum {
+          This = 0,
+          That = 1,
+        }
+        const rand = Math.random();
+        let theEnum: ExampleEnum | null = null;
+        if (rand < 0.3) {
+          theEnum = ExampleEnum.This;
+        }
```

Comment:
Nit: this can be simplified a bit 😎 

```suggestion
        const theEnum = Math.random () < 0.3 ? ExampleEnum.This : null;
```